src/Menge/MengeCore/Math/Geometry2D.h

```diff
@@ -747,19 +747,97 @@ namespace Menge {
 			 */
 			virtual Vector2 getCentroid() const;
 
-			/*!
-			 *	@brief		Returns the x-axis of the OBB's local frame
-			 *				
-			 *	@returns	The direction of the obb's x-axis.
-			 */
-			Vector2 getXBasis() const { return Vector2(_cosTheta, -_sinTheta); }
-
-			/*!
-			 *	@brief		Returns the y-axis of the OBB's local frame
-			 *				
-			 *	@returns	The direction of the obb's y-axis.
-			 */
-			Vector2 getYBasis() const { return Vector2(_sinTheta, _cosTheta); }
+      /*!
+      @brief    Returns the x-axis of the OBB's local frame expressed in the world frame.
+
+      If we say that Bx = getXBasis() and By = getYBasis(), are *column* vectors, we can define the 2x2
+      matrix R_WG = [Bx By] - a rotation taking vectors expressed in the geometry frame and expressing
+      them in the world frame.
+
+      For example, the position vector from the pivot to the far corner, expressed in the geometry
+      frame would be `r_PC_G = [w h]^T`. We can express that same vector in the world frame as:
+      `r_PC_W = R_WG * r_PC_G`.
+
+      Conversely, the matrix `R_GW = R_WG^T = [Bx By]^T` re-expresses in the *geometry* frame a vector
+      that was previously expressed in the world frame.
+
+      In practice, for a position vector from the world origin to point Q expressed in the world frame,
+      r_WQ_W, we can calculate a position vector from the geometry frame's origin r_WG (the OBB's
+      pivot) and expressed in the geometry frame as:
+      `r_GQ = [(r_WQ - r_WG) * Bx, (r_WQ - r_WG) * By]^T` or
+      `r_GQ = R_GW * (r_WQ - r_WG)`.
+      Functionality that becomes:
+
+      ```
+      Vector2 r_WQ_W;
+      Vector2 Bx = getXBasis();
+      Vector2 By = getYBasis();
+      Vector2 r_GQ_W = r_WQ - pivot_;
```

Comment:
This would be clearer if the definitions of `Bx` and `By` came *after* `r_GQ_W`. (Same for `getYBasis()`.)